casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -209,6 +210,7 @@ sealed abstract class SafetyOracleInstances {
           candidateMetadata: BlockMetadata,
           targetBlockHash: BlockHash
       ): F[Boolean] =
-        isInMainChain(blockDag, candidateMetadata, targetBlockHash)
+        ProtoUtil.getBlockMetadata[F](targetBlockHash, blockDag) >>=
+          (DagOperations.isDescendantOf[F](candidateMetadata, _, blockDag))
```

Comment:
I believe input args are misplaced here.
Should be `DagOperations.isDescendantOf[F](_, candidateMetadata, blockDag)`